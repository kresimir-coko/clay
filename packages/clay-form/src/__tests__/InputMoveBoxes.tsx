/**
 * SPDX-FileCopyrightText: Â© 2020 Liferay, Inc. <https://liferay.com>
 * SPDX-License-Identifier: BSD-3-Clause
 */

import {ClayInputMoveBoxes} from '..';
import {cleanup, render, fireEvent} from '@testing-library/react';
import React from 'react';

const options = [
	[
		{
			label: 'Option 1',
			value: '1',
		},
		{
			label: 'Option 2',
			value: '2',
		},
		{
			label: 'Option 3',
			value: '3',
		},
	],
	[
		{
			label: 'Option 4',
			value: '4',
		},
		{
			label: 'Option 5',
			value: '5',
		},
	],
];

describe('Rendering', () => {
	afterEach(cleanup);

	it('renders ClayInputMoveBoxes', () => {
		const {container} = render(
			<ClayInputMoveBoxes
				left={{
					items: options[0],
					label: 'In Use',
					onChange: () => {},
					onSelectChange: () => {},
					selected: [''],
				}}
				right={{
					items: options[1],
					label: 'Available',
					onChange: () => {},
					onSelectChange: () => {},
					selected: [''],
				}}
				size={8}
				spritemap="/path/to/some/resource.svg"
			/>
		);

		expect(container).toMatchSnapshot();
	});
});

describe('Interactions', () => {
	const handleOnItemsChange = jest.fn();

	afterEach(() => {
		handleOnItemsChange.mockReset();

		cleanup();
	});

	it('transfers selected options from left box to the right', () => {
		const {container} = render(
			<ClayInputMoveBoxes
				left={{
					items: options[0],
					label: 'In Use',
					onChange: handleOnItemsChange,
					onSelectChange: () => {},
					selected: ['1'],
				}}
				right={{
					items: options[1],
					label: 'Available',
					onChange: handleOnItemsChange,
					onSelectChange: () => {},
					selected: [''],
				}}
				size={8}
				spritemap="/path/to/some/resource.svg"
			/>
		);

		const transferRightButton = container.querySelector(
			'.transfer-button-ltr'
		);

		fireEvent.click(transferRightButton as HTMLObjectElement, {});

		expect(handleOnItemsChange).toHaveBeenCalled();

		const rightSelectBox = container.querySelector(
			'.select-box-right'
		) as HTMLSelectElement;

		// expect(rightSelectBox.options[2].value).toBe('1');
	});
});
