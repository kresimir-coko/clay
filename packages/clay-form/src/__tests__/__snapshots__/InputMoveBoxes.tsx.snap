// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders ClayInputMoveBoxes 1`] = `
<div>
  <div
    class="input-move-boxes"
  >
    <div
      class="form-group-autofit select-box"
    >
      <div
        class="form-group-item"
      >
        <label
          class="select-box-label"
        >
          In Use
        </label>
        <select
          class="select-box-left form-control select-box-select"
          multiple=""
          size="8"
        >
          <option
            class="select-box-option"
            value="1"
          >
            Option 1
          </option>
          <option
            class="select-box-option"
            value="2"
          >
            Option 2
          </option>
          <option
            class="select-box-option"
            value="3"
          >
            Option 3
          </option>
        </select>
        <div
          class="select-box-order-buttons btn-group"
          role="group"
        >
          <button
            class="select-box-order-button select-box-order-button-up btn btn-monospaced btn-secondary"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-caret-top"
              role="presentation"
            >
              <use
                xlink:href="/path/to/some/resource.svg#caret-top"
              />
            </svg>
          </button>
          <button
            class="select-box-order-button select-box-order-button-down btn btn-monospaced btn-secondary"
            type="button"
          >
            <svg
              class="lexicon-icon lexicon-icon-caret-bottom"
              role="presentation"
            >
              <use
                xlink:href="/path/to/some/resource.svg#caret-bottom"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="transfer-buttons btn-group"
      role="group"
    >
      <button
        class="transfer-button-ltr btn btn-monospaced btn-secondary"
        type="button"
      >
        <svg
          class="lexicon-icon lexicon-icon-caret-right"
          role="presentation"
        >
          <use
            xlink:href="/path/to/some/resource.svg#caret-right"
          />
        </svg>
      </button>
      <button
        class="transfer-button-rtl btn btn-monospaced btn-secondary"
        type="button"
      >
        <svg
          class="lexicon-icon lexicon-icon-caret-left"
          role="presentation"
        >
          <use
            xlink:href="/path/to/some/resource.svg#caret-left"
          />
        </svg>
      </button>
    </div>
    <div
      class="form-group-autofit select-box"
    >
      <div
        class="form-group-item"
      >
        <label
          class="select-box-label"
        >
          Available
        </label>
        <select
          class="select-box-right form-control select-box-select"
          multiple=""
          size="8"
        >
          <option
            class="select-box-option"
            value="4"
          >
            Option 4
          </option>
          <option
            class="select-box-option"
            value="5"
          >
            Option 5
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
